### ISSUE TITLE
keywords_case_sensitive test not pass

### ISSUE THREAD

**Issue by sundy-li** [vi3k6i5/flashtext#21]
see [https://github.com/vi3k6i5/flashtext/blob/master/test/keyword_extractor_test_cases.json#L92 ](url)

I am afraid the keywords_case_sensitive is not correct, and the test script did not check the keywords_case_sensitive cases.

**Comment by vi3k6i5** [comment]
Can you share the test case (A complete example)

Example: https://github.com/vi3k6i5/flashtext/issues/16

**Comment by sundy-li** [comment]
I modified the `test_extractor.py` and add this test, it was not passed
```
        for test_id, test_case in enumerate(self.test_cases):
            keyword_processor = KeywordProcessor(case_sensitive = True)
            keyword_processor.add_keywords_from_dict(test_case['keyword_dict'])
            keywords_extracted = keyword_processor.extract_keywords(test_case['sentence'])
            self.assertEqual(keywords_extracted, test_case['keywords_case_sensitive'],
                             "keywords_extracted don't match the expected results for test case: {}".format(test_id))
```

**Comment by vi3k6i5** [comment]
Awesome. Thanks, I will fix it soon :)

**Comment by sundy-li** [comment]
I just make a simple implementation in go (not fully), may be you could try the fixed test cases file 

[https://github.com/sundy-li/flashtext/blob/master/test_cases/extracts.json](url)
(It's json formatted)

**Comment by vi3k6i5** [comment]
Thanks, I realised that the code doesn't have bugs rather the test cases are buggy.
I have fixed it in develop branch. Will be pushing it out this weekend with the next version release.

Thanks again @sundy-li :)

**Comment by vi3k6i5** [comment]
Changing to enhancement as it's addition of more test case/coverage.

**Comment by vi3k6i5** [comment]
Fixed in flashtext-2.4.

Sorry I had to change the test case as well the json file. So I was not able to directly use your pull request. :(

But the bug is fixed. And lots of thanks for reporting it. 👍
